src/Donations.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { Component } from 'react';
+import { formatCurrency, getDonations, Donation, track } from './utils';
+
+type State = {
+  loading: boolean;
+  donations: Donation[];
+  totalAmount: number;
+};
+
+class Donations extends Component {
+  state: State = {
+    loading: true,
+    donations: [],
+    totalAmount: 0,
+  };
+
+  async componentDidMount() {
+    const donations = this.state.donations;
+
+    const result = await getDonations();
+
+    for (var i = 0; i < result.length; i++) {
+      donations.push(result[i]);
+    }
```

Comment:
Pointless - this is just starting with the state default empty array, and re-building the array returned from `getDonations`. Can be simplified to:
```suggestion
    const donations = await getDonations();
```